data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/DataPrepperArgs.java

```diff
@@ -22,11 +24,28 @@ public class DataPrepperArgs {
     private static final Integer MAXIMUM_SUPPORTED_NUMBER_OF_ARGS = 2;
 
     private static String checkForLogstashConfigurationAndConvert(String configurationFileLocation) {
-        if (configurationFileLocation.endsWith(".conf")) {
+        final File configurationLocationAsFile = new File(configurationFileLocation);
+        final LogstashConfigConverter logstashConfigConverter = new LogstashConfigConverter();
+        final Path configurationDirectory;
+
+        if (configurationLocationAsFile.isDirectory()) {
+            configurationDirectory = Paths.get(configurationFileLocation).toAbsolutePath();
+            FileFilter confFilter = pathname -> (pathname.getName().endsWith(".conf"));
+            for (File file : configurationLocationAsFile.listFiles(confFilter)) {
+                LOG.info("Detected logstash configuration file {}, attempting to convert to Data Prepper pipeline", file.getName());
+
+                try {
+                    logstashConfigConverter.convertLogstashConfigurationToPipeline(
```

Comment:
I suggest adding a javadoc comment to this function, explaining the return value can be a directory or file path with a reference to `PipelineParser.mergePipelineConfigurationFiles()`. Why this function sometimes returns an unmodified configurationFileLocation and other times a file path can be a little confusing.

Reply:
It is indeed a little confusing. Added some comment, but couldn't link to `PipelineParser.mergePipelineConfigurationFiles()` probably because it's a private method.